### DISCUSSION TITLE
Configuring Taxes by State/Province?

### DISCUSSION THREAD

**Discussion by alberbecois** [saleor/saleor#16094] [Q&A]
Hello,

I'm new to Saleor and am currently running it locally on Docker. I can't seem to find anywhere in the documentation or in the dashboard where I can set tax rates by province or state? I'm located in Canada and I will have to charge different taxes depending on the province. Can anybody explain how this works?

**Comment by tomaszlud** [comment]
Hey Alberbecois,

Saleor does have a marketplace integration with Avalara, which is designed for tax calculations. If you haven't tried it yet, it could be worth exploring. It is generally reliable for managing complex tax rules across various regions. 

You can also write an app from scratch, if possible, and, for example, use the Stripe API to calculate taxes. Let me know if it helps.

**Reply by alberbecois** [reply]
Hi,

Respectfully, this isn't really an acceptable answer. I did look into Avalara and there is zero pricing available upfront. I'm small art business owner trying to keep my costs low while I get started, which is why I'm looking at open source options. Saleor is supposed to be a complete solution and taxes are a critical part of running a business. I can't fathom why a site like Ko-fi is capable of letting you configure taxes by state or province and Saleor cannot. My question in response would be what niche is Saleor trying to fulfil if website builders like Squarespace or Wix are cheaper than an Avalara subscription? This is basic functionality and I would urge Saleor to consider implementing it.

**Reply by pkucmus** [reply]
Hi, @alberbecois I'm working with Saleor a lot and I'm here to share some knowledge I hope you'll find helpful.

Out-of-the box you have the tax framework in Saleor, there you can configure flat rates for countries / channels / tax classes - this is the level of granularity there is in the "flat rates mode". 

Saleor's advantage is that it's extendable and extending it means following an API contract (which is not the case in many other plug-in based platform). One can extend Saleor - in this case the tax calculation with a tax app. There might be a need of a custom one (something a developer would handle) or maybe the ones provided by Saleor is enough - the app provided with the entire context of the order can calculate the tax on the province level (or even deeper if that would ever make sense). Last time I checked there's an Avalara app and a TaxJar app (might be more) - maybe TaxJar will fit your needs if Avalara does not? 

I cannot reply to the "niche" question.

**Reply by timuric** [reply]
Hi, @alberbecois, sales tax in Canada can be tricky; even solutions like Wix would have Avalara integration for automated tax calculation.  If you prefer to do manual calculations, I advise you to look into tax calculation webhooks that you can use to provide custom business logic that works for you
